includes/class-wc-privacy.php

```diff
@@ -369,8 +369,59 @@ protected static function delete_inactive_accounts_query( $timestamp, $limit = 2
 				require_once ABSPATH . 'wp-admin/includes/user.php';
 			}
 
+			$builtin_post_types = get_post_types( array( '_builtin' => true ) );
+			unset( $builtin_post_types['user_request'] ); // Exclude user_request from data migration.
+
+			/**
+			 * Filter post types that should be assign to an admin user prior user exclusion.
+			 *
+			 * @param array $post_types List of post types to assign to an admin. Defaults to WordPress builtin post types.
+			 */
+			$post_types_to_assign = apply_filters( 'woocommerce_delete_inactive_account_post_types_to_assign', array_values( $builtin_post_types ) );
+
 			foreach ( $user_ids as $user_id ) {
-				wp_delete_user( $user_id );
+				$posts_made_prior = get_posts(
+					array(
+						'post_type'   => $post_types_to_assign,
+						'author'      => $user_id,
+						'numberposts' => -1,
+					)
+				);
+
+				if ( empty( $posts_made_prior ) ) {
+					wp_delete_user( $user_id );
+				} else {
+					$admin_users = get_users( array( 'role' => 'administrator' ) );
```

Comment:
`wp_delete_user()` already supports to reassign to a new user, so we could save some lines of code.

See: https://developer.wordpress.org/reference/functions/wp_delete_user/

So we could change to:

```
$admin_users = get_users( array( 'role' => 'administrator' ) );
$deleted_user = get_user_by( 'ID', $user_id );
wp_delete_user( $user_id, $admin_users[0]->ID ); // Delete user and reassign content to admin.

$posts_made_prior_list = '<ul>';
foreach ( $posts_made_prior as $post ) {
    $posts_made_prior_list .= '<li><a href="' . esc_attr( $post->guid ) . '">' . esc_html( $post->post_title ) . '</a></li>';
}
$posts_made_prior_list .= '</ul>';
```

Reply:
Done! :)